### PR TITLE
task was solved

### PR DESCRIPTION
**Pull request by vanyachyzh**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,37 @@
...
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      for (const item in actions[i].extraData) {
+        state[item] = actions[i].extraData[item];
+      }
+
+      resultArray.push({
+        ...state,
+      });
```
Comment (IvanFesenko) [inline src/transformState.js:18]: Do not repeat your self, do this action once at the end

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Good job. Please fix code duplicating

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
```
Comment (AlisaVlasova) [inline src/transformState.js:11]: use switch/case

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,29 @@
...
+      for (const item in actions[i].extraData) {
```
Comment (AlisaVlasova) [inline src/transformState.js:12]: use Object.assign

**Review by AlisaVlasova — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 4]
```diff
@@ -5,7 +5,33 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (true) {
+      case actions[i].type === 'addProperties':
+        Object.assign(state, actions[i].extraData);
```
Comment (OlEzhA0) [inline src/transformState.js:13]: don't mutate input state, add some constant

**Review by OlEzhA0 — CHANGES_REQUESTED** [review]

File: src/transformState.js (hunk 4 above)
Comment (vanyachyzh) [reply src/transformState.js:13]: - `state` is an object. You are supposed to add, change, or delete its
  properties instead of creating a new object

Comment (vanyachyzh) [reply src/transformState.js:13]: I think I have to mutate state, don't I?

File: src/transformState.js [hunk 5]
```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
```
Comment (IvanFesenko) [inline src/transformState.js:11]: for of loop will be better 
```suggestion
  for (const action of actions) {
    switch (actions.type) {
```

File: src/transformState.js [hunk 6]
```diff
@@ -5,7 +5,31 @@
...
+      case 'addProperties':
+        Object.assign(state, actions[i].extraData);
+        break;
+
+      case 'removeProperties':
+        for (let j = 0; j < actions[i].keysToRemove.length; j++) {
+          delete state[actions[i].keysToRemove[j]];
```
Comment (IvanFesenko) [inline src/transformState.js:18]: Same here
```suggestion
       for (const key of action.keysToRemove) {
         delete state[key];
```

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Good job. Let's make some improvements

**Review by etojeDenys — APPROVED** [review]
LGTM!

**Pull request closed without merging**